### DISCUSSION TITLE
more than 2 worker server to run with worker

### DISCUSSION THREAD

**Discussion by yearcode** [celery/celery#9211] [Q&A]
hi all:
     i have a task, and set it with queue; i have  more than 2 worker server, when i start worker(every server to run   2 commands)
```
celery -A MyService worker -Q job_1  --concurrency=10    ......
celery -A MyService worker --concurrency=20 ..........
```
i have 2 question，
question 1:
      more then 1 server to run queue job_1, how many concurrency to run job_1(only 10 or concurrency*len(servers))? i want to only 10 concurrency to run queue job_1's task

question 2:
     i start queue job_1 and start  worker,  like this:
```
celery -A MyService worker -Q job_1  --concurrency=10    ......
celery -A MyService worker --concurrency=20 ..........
```

the second command would execute queue job_1 tasks?

thanks!

**Comment by gyanu2507** [comment]
Concurrency is per worker process, not a cluster cap.

Two servers both running `celery -A MyService worker -Q job_1 --concurrency=10` give **20** slots for `job_1`. There is no built-in "only 10 across every host". To keep a global 10 you run one worker with `--concurrency=10`, or use a rate limit / a lock inside the task.

The second command, `celery -A MyService worker --concurrency=20` with no `-Q`, consumes the default queue (`celery`). It will not take `job_1` messages unless you routed those tasks to `celery` as well.

So: job_1 tasks stay on the `-Q job_1` workers. The unnamed worker is a separate pool on the default queue.
